interview-questions.md

```diff
@@ -9,47 +9,83 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is yarn? What file(s) are modified when you run the command yarn in your terminal?
 
 Your answer:
+Yarn is a terminal command that allows the user to pull in the necessary extentions to perform tasks like testing code with jest. By using the command yarn jest you can run the added files in your code to execute test that will either fail or pass based on the funciton you created. 
 
 Researched answer:
+Yarn is a very common package manager for JavaScript and stands for Yet Another Resource Negotiator. Package .json is a file that is ran by the comman yarn. It contains functional metadata(name, version, and dependencies).
+
+answer via:
+https://phoenixnap.com/kb/package-json#:~:text=npm%20init%20command.-,The%20package.,the%20project%20and%20handle%20dependencies.
+
+
 
 3. What is a React component?
 
 Your answer:
+A react component is an individual piece of a webpage. For example if we are writing code in react and want to build a navigation bar for users we can place that in a seperate file. From here we would import that file to our main file app.js as our brains of the code. This helps the program to be more organized and concise.
 
 Researched answer:
+Components are reusable bits of code that are seperated from eachother. Their are two types of componets, class and function componets. They act like JavaScript functions, however are orgranized separately.
+
+answer via:
+https://www.w3schools.com/react/react_components.asp#:~:text=Components%20are%20independent%20and%20reusable,will%20concentrate%20on%20Function%20components.
 
 4. What is the difference between state values and props in React?
 
 Your answer:
+State values are a way to make our JSK code dynamic. They are a variable that is declared to be used throughout our code that will allow the value to changed by user input . Props is short for properties. They act much like parameters for our funcitons that allow them to have different behavor to then be called and displayed on the webpage. 
 
 Researched answer:
+The difference between state values and props is that state is locally owned and is update by the componet. Props are only read by their parent. Their change is based on the upstrean callback function. 
 
-5. What is the DOM?
+answer via:
+https://www.knowledgehut.com/blog/web-development/react-state-vs-props
 
-Your answer:
+
+5. What is the DOM?
+Your answer: DOM stands for (forgot what the D was for) OBJECT Model. It is how the flow of events takes place from embedded funcitonalty to what the user intereacts with real-time. 
 
 Researched answer:
+The Documnet Object Model is the intereface for the web document/page. It is the structure that represents the webpage so that things such as style, elemets, and functionality can be updated. s
+
+answer via:
+https://developer.mozilla.org/en-US/docs/Web/API/Document_Object_Model/Introduction
 
```

Comment:
Awesome job, I can see improvement! I encourage you to use examples whenever you can especially in your research. It will help solidify your understanding.